src/components/create-post.jsx

```diff
@@ -224,3 +224,51 @@ export default class CreatePost extends React.Component {
     );
   }
 }
+
+/**
+ * Convert 'image/png' blob to 'image/jpeg' blob if:
+ * 1) The PNG size is more than 100 KiB and
+ * 2) JPEG size is less than half the PNG size.
+ *
+ * The returning promise is never failed and returns
+ * either JPEG or the original PNG.
+ *
+ * @param {Blob} blob
+ * @returns {Promise<Blob>}
+ */
+export async function makeJpegIfNeeded(blob) {
+  if (blob.type !== 'image/png' || blob.size < 100 * 1024) {
```

Comment:
why 100KiB limit? difference can still be pretty dramatic for smaller images

Reply:
I thought that <100 KiB PNG won't hurt anyone, even if he's not optimal. We can lower this limit but i think it should be more than zero to keep some small but colored images (icons, drawed userpics) as PNG.

Reply:
> <100 KiB PNG won't hurt anyone
Is it "PNG" though? I mean, is blob size equals PNG size?

Reply:
Yes, the pasted Blob _is_ a PNG-formatted file made by browser or OS (I dont know for sure). It is the same PNG that will be uploaded to our server.